test/language/generators/syntax/yield-as-function-expression-binding-identifier.js

```diff
@@ -0,0 +1,14 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+  description: >
+      `yield` is a reserved keyword within generator function bodies and may
+      not be used as the binding identifier of a function expression.
+  es6id: 12.1.1
+  negative: SyntaxError
+ ---*/
+
+function* g() {
+  (function yield() {})
```

Comment:
Got it! You can follow along here if you like: [Issue 3983: Incorrect restriction on binding identifiers of function expressions](https://code.google.com/p/v8/issues/detail?id=3983)